mobile_verifier/src/reward_share.rs

```diff
@@ -0,0 +1,312 @@
+use chrono::{DateTime, Duration, TimeZone, Utc};
+use futures::stream::StreamExt;
+use helium_proto::services::{
+    follower::{self, FollowerGatewayReqV1},
+    poc_mobile::{Share as ShareProto, ShareValidity},
+    Channel,
+};
+use lazy_static::lazy_static;
+use poc_store::{
+    heartbeat::{CellHeartbeat, CellHeartbeatIngestReport},
+    speedtest::{CellSpeedtest, CellSpeedtestIngestReport},
+    traits::MsgDecode,
+    BytesMutStream,
+};
+use rust_decimal::Decimal;
+use rust_decimal_macros::dec;
+use serde::Serialize;
+
+use crate::{
+    cell_type::CellType,
+    mobile::Mobile,
+    reward_speed_share::{InvalidSpeedShares, SpeedShare, SpeedShareMovingAvgs, SpeedShares},
+    Result,
+};
+use helium_crypto::PublicKey;
+use std::collections::HashMap;
+
+/// Map from cbsd_id to share
+pub type Shares = HashMap<String, Share>;
+
+/// Map from cbsd_id to invalid share
+pub type InvalidShares = Vec<ShareProto>;
+
+/// Map from cell_type to accumulated_reward_weight (decimal)
+pub type CellShares = HashMap<CellType, Decimal>;
+
+/// Map from gw_public_key to accumulated_reward_weight (decimal)
+pub type HotspotShares = HashMap<PublicKey, Decimal>;
+
+/// Map from owner_public_key to accumulated_reward_weight (decimal)
+pub type OwnerShares = HashMap<PublicKey, Decimal>;
+
+/// Map from gw_public_key (without owners) to accumulated_reward_weight (decimal)
+pub type MissingOwnerShares = HashMap<PublicKey, Decimal>;
+
+/// Map from owner_public_key to accumulated_rewards (mobile)
+#[derive(Debug, Clone, Serialize)]
+pub struct OwnerEmissions(HashMap<PublicKey, Mobile>);
+
+// 100M genesis rewards per day
+const GENESIS_REWARDS_PER_DAY: i64 = 100_000_000;
+
+lazy_static! {
+    static ref GENESIS_START: DateTime<Utc> = Utc.ymd(2022, 7, 11).and_hms(0, 0, 0);
+}
+
+impl OwnerEmissions {
+    pub fn new(owner_shares: OwnerShares, start: DateTime<Utc>, duration: Duration) -> Self {
+        let mut owner_emissions = HashMap::new();
+        let total_shares: Decimal = owner_shares.values().sum();
+        if total_shares > dec!(0) {
+            if let Some(actual_emissions) = get_scheduled_tokens(start, duration) {
+                let emissions_per_share = actual_emissions / total_shares;
+                for (owner, share) in owner_shares {
+                    owner_emissions.insert(owner, Mobile::from(share * emissions_per_share));
+                }
+            }
+        }
+        OwnerEmissions(owner_emissions)
+    }
+
+    #[allow(dead_code)]
```

Comment:
This still dead?

Reply:
It is not, but the function is only used in tests so this still needs to be here :-/

Reply:
Would it be better to just make it a public function or move it to a test-only function if it’s being called in multiple modules’ tests?

Reply:
It is public, we'd have to make OwnerEmissions pub and reachable from the root of the module as well